File: 3702-longest-subsequence-with-non-zero-bitwise-xor/3702-longest-subsequence-with-non-zero-bitwise-xor.cpp
class Solution {
public:
    int longestSubsequence(vector<int>& nums) {
        int n = nums.size();

        int resxor = 0;
        bool zeros = true;

        for (int &x : nums){
            resxor ^= x;

            if (x != 0) zeros=false;
        }

        if (zeros) return 0;

        return resxor==0 ? n-1 : n;
    }
};

File: 3702-longest-subsequence-with-non-zero-bitwise-xor/3702-longest-subsequence-with-non-zero-bitwise-xor.py
class Solution:
    def longestSubsequence(self, nums: List[int]) -> int:
        resxor = 0
        n = len(nums)
        allzero = True

        for x in nums:
            resxor ^= x

            if x != 0: allzero=False #at least one zero found 
        
        if allzero: return 0

        return n if resxor!=0 else n-1